### PR TITLE
ref(page-filters): Use counter badge & better trimming in project filter

### PR DESCRIPTION
**Pull request by vuluongj20**
* Limit the number of shown projects to 2, and show a counter badge when user selects more than 2 projects.
    Screen Shot 2022-03-28 at 10 33 26 AM
    Screen Shot 2022-03-28 at 10 33 42 AM
* Use a custom trimming function to trim long project names from the middle. This is useful because a lot of projects names contain important information at the end, e.g. `symbol-collector-mobile` and `symbol-collector-server`. 
    Screen Shot 2022-03-28 at 10 39 41 AM
    Screen Shot 2022-03-28 at 10 34 55 AM

### CODE REVIEW

File: static/app/components/badge.tsx [hunk 1]
```diff
@@ -27,7 +27,6 @@ const Badge = styled(({children, text, ...props}: Props) => (
   transition: background 100ms linear;
 
   position: relative;
-  top: -1px;
```
Comment (vuluongj20) [inline static/app/components/badge.tsx:30]: This was added to better align `<Badge />` inside nav tabs.

With `top: -1px`:
Screen Shot 2022-03-28 at 1 31 15 PM

Without:
Screen Shot 2022-03-28 at 1 31 30 PM

However, this results in `<Badge />` being misaligned everywhere else. Better to remove this and center-align the content inside nav tabs (https://github.com/getsentry/sentry/pull/33002/commits/aeb6b7aa4284cc1d061c7f28eb48b8d3c9fb2482).

File: static/less/shared-components.less [hunk 2]
```diff
@@ -1112,6 +1112,8 @@ header + .alert {
     }
 
     > a {
+      display: flex;
+      align-items: center;
```
Comment (vuluongj20) [inline static/less/shared-components.less:1116]: Before:
Screen Shot 2022-03-28 at 1 35 45 PM

After:
Screen Shot 2022-03-28 at 1 36 05 PM

File: static/app/components/badge.tsx (hunk 1 above)
Comment (evanpurkhiser) [reply static/app/components/badge.tsx:30]: 👍  nice

File: static/app/utils.tsx [hunk 3]
```diff
@@ -120,6 +120,56 @@ export function explodeSlug(slug: string): string {
...
 }
 
+/**
+ * Trim slug name with a preference for preserving whole words. Only cut up
+ * whole words if the last remaining words are still too long. For example:
+ * "javascript-project-backend" --> "javascript…backend"
+ * "my-long-sentry-project-name" --> "my-long-sentry…name"
+ * "javascriptproject-backend" --> "javascriptproje…ckend"
+ */
+export function trimSlug(slug: string, maxLength: number = 20) {
```
Comment (evanpurkhiser) [inline static/app/utils.tsx:130]: I think we can probably call this just `trimMiddle`

Comment (evanpurkhiser) [reply static/app/utils.tsx:130]: Oh I guess it is a bit special though, that it splits on `-`s

Comment (evanpurkhiser) [reply static/app/utils.tsx:130]: Can we write some unit tests for this?

**Review by evanpurkhiser — COMMENTED** [review]
Loving this. Just need some tests on the util

File: static/app/utils.tsx (hunk 3 above)
Comment (evanpurkhiser) [reply static/app/utils.tsx:130]: Also let's move it to it's own file `static/app/utils/trimSlug.tsx`

Comment (vuluongj20) [reply static/app/utils.tsx]: Ahh good points! Updated in https://github.com/getsentry/sentry/pull/33002/commits/254da1f7ffd990b862ae30ad54cb0834e14657ed

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**